src/main/java/net/snowflake/ingest/streaming/example/SnowflakeStreamingIngestExample.java

```diff
@@ -35,28 +39,55 @@ public static void main(String[] args) throws Exception {
     // Create a streaming ingest client
     try (SnowflakeStreamingIngestClient client =
         SnowflakeStreamingIngestClientFactory.builder("CLIENT").setProperties(props).build()) {
+
       // Create an open channel request on table T_STREAMINGINGEST
       OpenChannelRequest request1 =
-          OpenChannelRequest.builder("CHANNEL")
-              .setDBName("DB_STREAMINGINGEST")
-              .setSchemaName("SCHEMA_STREAMINGINGEST")
-              .setTableName("T_STREAMINGINGEST")
+          OpenChannelRequest.builder("MY_CHANNEL")
+              .setDBName("MY_DATABASE")
+              .setSchemaName("MY_SCHEMA")
+              .setTableName("MY_TABLE")
               .setOnErrorOption(OpenChannelRequest.OnErrorOption.CONTINUE)
               .build();
 
       // Open a streaming ingest channel from the given client
       SnowflakeStreamingIngestChannel channel1 = client.openChannel(request1);
 
-      // Insert a few rows into the channel,
-      for (int val = 0; val < 1000; val++) {
+      // Insert few rows into the channel (Using insertRows API), along with the offset Token which
+      // corresponds to the row number
+      final int totalRowsInTable = 1000;
+      for (int val = 0; val < totalRowsInTable; val++) {
         Map<String, Object> row = new HashMap<>();
+
+        // c1 corresponds to the column name in table
         row.put("c1", val);
-        InsertValidationResponse response = channel1.insertRow(row, null);
+
+        InsertValidationResponse response = channel1.insertRow(row, String.valueOf(val));
         if (response.hasErrors()) {
           // Simply throw if there is an exception
           throw response.getInsertErrors().get(0).getException();
         }
       }
+
+      // Polling Snowflake to fetch offset token registered in Snowflake
+      final int expectedOffsetTokenInSnowflake = 999; // because it goes from 0 to 999
+      String offsetTokenFromSnowflake = channel1.getLatestCommittedOffsetToken();
+
+      final int maxRetries = 10;
+      int retryCount = 0;
+      while (offsetTokenFromSnowflake == null
+          || !offsetTokenFromSnowflake.equals(String.valueOf(expectedOffsetTokenInSnowflake))) {
+        Thread.sleep(1_000);
+        offsetTokenFromSnowflake = channel1.getLatestCommittedOffsetToken();
+        retryCount++;
+        if (retryCount >= maxRetries) {
+          System.out.println(
+              String.format(
+                  "Failed to look for required OffsetToken in Snowflake:%s after MaxRetryCounts:%s",
+                  expectedOffsetTokenInSnowflake, maxRetries));
+          System.exit(1);
+        }
+      }
+      System.out.println("SUCCESSFULLY inserted " + totalRowsInTable + " rows");
```

Comment:
This is exactly what we do in `channel.close`, could we replace with it and just explain as a comment?